generate/testdata/snapshots/TestGenerate-UnionNoFragments.graphql-UnionNoFragments.graphql.go

```diff
@@ -21,9 +22,35 @@ type UnionNoFragmentsQueryRandomLeafLeafContent interface {
 	implementsGraphQLInterfaceUnionNoFragmentsQueryRandomLeafLeafContent()
 }
 
-func (v UnionNoFragmentsQueryRandomLeafArticle) implementsGraphQLInterfaceUnionNoFragmentsQueryRandomLeafLeafContent() {
+func (v *UnionNoFragmentsQueryRandomLeafArticle) implementsGraphQLInterfaceUnionNoFragmentsQueryRandomLeafLeafContent() {
 }
-func (v UnionNoFragmentsQueryRandomLeafVideo) implementsGraphQLInterfaceUnionNoFragmentsQueryRandomLeafLeafContent() {
+func (v *UnionNoFragmentsQueryRandomLeafVideo) implementsGraphQLInterfaceUnionNoFragmentsQueryRandomLeafLeafContent() {
+}
+
+func __unmarshalUnionNoFragmentsQueryRandomLeafLeafContent(v *UnionNoFragmentsQueryRandomLeafLeafContent, m json.RawMessage) error {
```

Comment:
Is there an advantage of using `v` as an out argument instead of returning `(InterfaceNestingRootTopicChildrenArticleParentTopicChildrenContent, error)`?

Then the code below would be:
```
err, v.RandomLeaf = __unmarshalUnionNoFragmentsQueryRandomLeafLeafContent(firstPass.RandomLeaf)
```

Reply:
Ah, interesting.  I was doing this to match json.Unmarshal, and I do think it's marginally more efficient this way, although probably not to an extent that matters.

But I think actually the biggest factor is what will make the most sense when list-of-interface fields make the caller more complicated (#54).  I think actually the out-argument makes things simpler; but maybe you can see what you think when you get there.